app/scripts/modules/amazon/src/domain/IAmazonLaunchTemplate.ts

```diff
@@ -0,0 +1,87 @@
+import { IBlockDeviceMapping } from './IAmazonBlockDeviceMapping';
+
+export interface IIamInstanceProfile {
+  arn?: string;
+  name: string;
+}
+
+export interface ICpuOptions {
+  coreCount?: number;
+  threadsPerCore?: number;
+}
+
+export interface IElasticGpuSpecification {
+  type?: string;
+}
+
+export interface IElasticInterfaceAccelerator {
+  count?: number;
+  type?: string;
+}
+
+export interface ISpotMarketOptions {
+  blockDurationMinutes?: number;
+  instanceInterruptionBehavior?: string;
+  maxPrice?: string;
+  spotInstanceType?: 'one-time' | 'persistent';
+  validUntil?: string;
+}
+
+export interface IInstanceMarketOptions {
+  marketType?: string;
+  spotOptions?: ISpotMarketOptions;
+}
+
+export interface ILicenseConfig {
+  licenseConfigurationArn?: string;
+}
+
+export interface IMetadataOptions {
+  httpEndpoint?: 'disabled' | 'enabled';
+  httpPutResponseHopLimit?: number;
+  httpsTokens?: 'required' | 'optional';
+  state?: 'pending' | 'applied';
+}
+
+export interface ITagSpecification {
+  resourceType?: string;
+  tagSet?: Array<{
+    [key: string]: string;
```

Comment:
Wondering if this will be confusing to people down the road.  An array of tag objects, but each tag object can only contain a single key: value pair?  Should we use the `Tag` representation that Amazon documents instead?

```
{
  "Key" : String,
  "Value" : String
}
```